pkg/util/deployment/deployment.go

```diff
@@ -53,9 +55,12 @@ func GetOldReplicaSets(deployment extensions.Deployment, c clientset.Interface)
 		})
 }
 
+type rsListFunc func(string, api.ListOptions) ([]extensions.ReplicaSet, error)
```

Comment:
Note that we have also been special-casing interfaces in our deployment controllers and now we are looking into switching back to full namespacers since they are much cleaner and more predictable imho. Why can't we use namespacers here too?